lib/src/execution/windows/execution.cc

```diff
@@ -1,578 +0,0 @@
-
-    void log_execution(string const& file, vector<string> const* arguments);
-
-    const char *const command_shell = "cmd.exe";
-    const char *const command_args = "/c";
-
-    struct extpath_helper
-    {
-        vector<string> const& ext_paths() const
-        {
-            return _extpaths;
-        }
-
-        bool contains(const string & ext) const
-        {
-            return binary_search(_extpaths.begin(), _extpaths.end(), to_lower_copy(ext));
-        }
-
-     private:
-        // Use sorted, lower-case operations to ignore case and use binary search.
-        vector<string> _extpaths = {".bat", ".cmd", ".com", ".exe"};;
-    };
-
-    static bool is_executable(path const& p, extpath_helper const* helper = nullptr)
-    {
-        // If there's an error accessing file status, we assume is_executable
-        // is false and return. The reason for failure doesn't matter to us.
-        boost::system::error_code ec;
-        bool isfile = is_regular_file(p, ec);
-        if (ec) {
-            LOG_TRACE("error reading status of path %1%: %2% (%3%)", p, ec.message(), ec.value());
-        }
-
-        if (helper) {
-            // Checking extensions aren't needed if we explicitly specified it.
-            // If helper was passed, then we haven't and should check the ext.
-            isfile &= helper->contains(p.extension().string());
-        }
-        return isfile;
-    }
-
-    string which(string const& file, vector<string> const& directories)
-    {
-        // On Windows, everything has execute permission; Ruby determined
-        // executability based on extension {com, exe, bat, cmd}. We'll do the
-        // same check here using extpath_helper.
-        static extpath_helper helper;
-
-        // If the file is already absolute, return it if it's executable.
-        path p = file;
-        if (p.is_absolute()) {
-            return is_executable(p, &helper) ? p.string() : string();
-        }
-
-        // On Windows, treat 'echo' as a command that can be found
-        if (file == "echo") {
-            return "echo";
-        }
-
-        // Otherwise, check for an executable file under the given search paths
-        for (auto const& dir : directories) {
-            path p = path(dir) / file;
-            if (!p.has_extension()) {
-                path pext = p;
-                for (auto const&ext : helper.ext_paths()) {
-                    pext.replace_extension(ext);
-                    if (is_executable(pext)) {
-                        return pext.string();
-                    }
-                }
-            }
-            if (is_executable(p, &helper)) {
-                return p.string();
-            }
-        }
-        return {};
-    }
-
-    // Create a pipe, throwing if there's an error. Returns {read, write} handles.
-    static tuple<scoped_resource<HANDLE>, scoped_resource<HANDLE>> CreatePipeThrow(DWORD read_mode = 0, DWORD write_mode = 0)
-    {
-        static LONG counter = 0;
```

Comment:
We could create the name based off a timestamp, like GetTickCount64. Probably still have potential for identical numbers if running in parallel.